### PR TITLE
IBM LLM initiated

### PR DESCRIPTION
**Pull request by Balamurugan35**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ def requirements():
         "auto-gptq==0.5.0",
         "openai==0.28.1",
         "uvicorn==0.24.0.post1",
-        "tiktoken==0.5.1"
+        "tiktoken==0.5.1",
+        "ibm-watson-machine-learning==1.0.327"
     ]
```
Comment (SyedHabibM) [inline setup.py:36]: version should be changed/updated?!

**Review by SyedHabibM — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (Balamurugan35) [reply setup.py:36]: New package added

File: semantic_ai/llm/ibm.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            raise ValueError(
+                f"{self.model_type} not in the model type list. Please give the following model type: {model_type_list}"
+            )
+
+    async def llm_model(self):
+        ibm_llm_project_id = self.project_id
+        ibm_llm_model = Model(
+            model_id=self.model_type,
+            credentials=self.ibm_llm_credentials,
+            project_id=ibm_llm_project_id,
```
Comment (SyedHabibM) [inline semantic_ai/llm/ibm.py:65]: We can directly use `self.project_id` here.

**Review by SyedHabibM — CHANGES_REQUESTED** [review]

**Review by SyedHabibM — APPROVED** [review]

**Comment by SyedHabibM** [comment]
@PrabhuRaghav Plz verify and approve.

**Review by PrabhuRaghav — APPROVED** [review]

**Comment by PrabhuRaghav** [comment]
> @PrabhuRaghav Plz verify and approve.

@SyedHabibM PRs reviewed and approved

**Pull request merged**